res/controllers/midi-components-0.0.js

```diff
@@ -27,14 +27,63 @@
  */
 
 (function(global) {
+    /**
+     * @typedef {[number, number] | [number, [number, number]] | [[number, number], number] | [[number, number], [number, number]]} MidiBytesVariations
+     */
+
+    /**
+     * @typedef {object} MidiBytesAsObject
+     * @property {MidiBytesVariations} [in] MIDI signals from device to listen
+     * @property {MidiBytesVariations} [out] MIDI signals to transmit to the device
+     */
+
+    /**
+     * @typedef {object} ComponentOptions
+     * @property {MidiBytesAsObject | [number, number]} [midi] MIDI events
+     */
+
     const NO_TIMER = 0;
+    /**
+     * @param {ComponentOptions | [number, number]} options Component configuration
+     * @class
+     */
     const Component = function(options) {
-        if (Array.isArray(options) && typeof options[0] === "number") {
-            this.midi = options;
+        if (Array.isArray(options)) {
+            const [midiStatus, midiNo] = options;
+            options = {midi: {out: [midiStatus, midiNo]}};
+        } else if (Array.isArray(options.midi)) {
+            options.midi = {out: options.midi};
```

Comment:
Oh, I actually just noticed something. There are probably other mappings out there that assume `midi` to have the same shape as it had previously. I don't think we can just change it like that. Otherwise custom `send` methods will fail when they do stuff like `midi.sendShortMessage(this.midi[0], this.midi[1], value)`...